### PR TITLE
Acorn Tracking

### PR DESCRIPTION
**Pull request by UvuvDev**
#### Summary

Added Acorn Tracking, derived from movement class. It takes in a vision sensor and a vision sensor signature as parameters. Gets global coordinates of acorns 

#### Motivation

ion know

#### Test Plan

made an app and tested it with opencv and my camera

#### Additional Notes

none

## Download the template for this pull request: 

> [!NOTE]  
> This is auto generated from [`Add Template to Pull Request`](https://github.com/LemLib/LemLib/actions/runs/7510387748)
- via manual download: [LemLib@0.5.0-rc2+910f9e.zip](https://nightly.link/LemLib/LemLib/actions/artifacts/1166935943.zip)
- via PROS Integrated Terminal: 
 ```
curl -o LemLib@0.5.0-rc2+910f9e.zip https://nightly.link/LemLib/LemLib/actions/artifacts/1166935943.zip;
pros c fetch LemLib@0.5.0-rc2+910f9e.zip;
pros c apply LemLib@0.5.0-rc2+910f9e;
rm LemLib@0.5.0-rc2+910f9e.zip;
```

### CODE REVIEW

**Comment by liam-teale** [comment]
wrong branch. PR it into the refactor branch

**Comment by liam-teale** [comment]
is this ready for testing?

**Comment by UvuvDev** [comment]
> is this ready for testing?

Should be, just check the old UvuvLib documentation on how the distance to height thing works. Or dm me, either is fine

File: src/lemlib/movements/acorntracking.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                y1 = 0;
+                x2 = VISION_FOV_WIDTH;
+                y2 = 0;                  
+            }
+
+        }
+
+        float distance = y1 + ((trackedAcornRadius-x1) * (y2-y1))/ (x2-x1);
+        // Linear Interpolation: y1 + [(x-x1) * (y2-y1)]/ (x2-x1)
+        std::pair<float, float> newcoords = std::make_pair<float, float>(pose.x + distance*std::cos(pose.theta), pose.y + distance*std::sin(pose.theta));
```
Comment (meisZWFLZ) [inline src/lemlib/movements/acorntracking.cpp]: The output of the movement::update() is intended to be the power of the left and right sides of the chassis..

**Pull request closed without merging**